File: C++/프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.cpp
#include <string>
#include <vector>
// #include <iostream>

using namespace std;

string solution(int n, int t, int m, int p) {
    vector<char> numlist = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F'};
    string wholeStr = "0";
    int num = 0, tmp = 0;
    
    string strNum;
    while(wholeStr.size() / m < t) {
        strNum = "";
        tmp = num++;
        while(0 < tmp) {
            strNum = numlist[tmp % n] + strNum;
            tmp /= n;
        }
        wholeStr += strNum;
    }
    
    // cout << "Whole String: " << wholeStr << endl;
    string answer = "";
    --p;
    while(answer.size() < t) {
        answer += wholeStr[p];
        p += m;
    }
    
    return answer;
}

File: Python3/프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.py
def solution(n, t, m, p):
    answer = ''
    
    numlist = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F']
    str_num = ""
    whole_str = "0"
    for i in range(m * t):
        str_num = ""
        while 0 < i:
            str_num = numlist[i % n] + str_num
            i //= n
        whole_str += str_num
        
    return whole_str[p - 1:len(whole_str):m][:t]